kustomize/base/kustomization.yaml

```diff
@@ -16,15 +16,4 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 
 resources:
-  - adservice.yaml
-  - cartservice.yaml
-  - checkoutservice.yaml
-  - currencyservice.yaml
-  - emailservice.yaml
-  - frontend.yaml
-  - loadgenerator.yaml
-  - paymentservice.yaml
-  - productcatalogservice.yaml
-  - recommendationservice.yaml
-  - redis.yaml
-  - shippingservice.yaml
+  - kubernetes-manifests.yaml
```

Comment:
**Issue:**
One reason we adopted Kustomize is so we could move away from using a single, all-encompassing `kubernetes-manifests.yaml` file. 
Let's showcase best practices by using multiple/separate YAML files (i.e., `adservice.yaml`, `cartservice.yaml`, etc.).

We could introduce a new bash function similar to [`mk_kubernetes_manifests`](https://github.com/GoogleCloudPlatform/microservices-demo/blob/release/v0.3.9/hack/make-release-artifacts.sh#L63) that creates these individual files. In that function, we:
1. Go through each YAML file in the [`/kubernetes-manifests` folder](https://github.com/GoogleCloudPlatform/microservices-demo/tree/main/kubernetes-manifests).
    1. Copy the file to `/kustomize/base`.
    2. Replace Docker service image(s) — similar to [`mk_kubernetes_manifests`](https://github.com/GoogleCloudPlatform/microservices-demo/blob/release/v0.3.9/hack/make-release-artifacts.sh#L68).
    3. [Optional] Region tag. This is not a priority.
We can call the function `mk_kustomize_base`.

What do you think?
This is a lot of work, so I am happy to do it myself in the coming weeks. 😊

Reply:
I see what you mean, but to be honest this `kustomize/base/` folder doesn't have any sense but just being the copy of the `release/kubernetes-manifests.yaml` file. Having the same file/format between the 2 folders is very convenient indeed in the release process I updated (note: this `kustomize/base/*` files were already out-dated).

I would prefer if you'll do this approach of splitting (I'm not sure we need to do that, but that's your call ;)) the file in multiple files in another PR, like you proposed. Deal?

Reply:
Thank you for the quick response, Mathieu!

**Value of Splitting**
You bring up a good point: the value of splitting `/kustomize/base/kubernetes-manifests.yaml` _might_ not be worth the effort of creating a `mk_kustomize_base` function. Here's the value I can think of:
- `/kustomize/base` will eventually be the default place (**ideal path**) users should use to deploy — so it's should showcase **best practices**. Splitting is better practice than one, large file.
- Using separate files is more readable and approachable to new users. As a new users, looking at the will quickly tell you roughly how many services will be deployed.
- We can still deploy via a single command without `git clone`-ing this repo — by using [Kustomize remote targets](https://github.com/kubernetes-sigs/kustomize/blob/master/examples/remoteBuild.md). So we're not losing convenience.
- There's been interest in the past for moving away from the use of a single, all-encapsulating `kubernetes-manifests.yaml` file (in https://github.com/GoogleCloudPlatform/microservices-demo/issues/375).

**Out-of-scope for this PR**
Since temporarily using a `/kustomize/base/kubernetes-manifests.yaml` won't break anything, and since this PR contains an important CI improvement, I'm happy merge this PR with `/kustomize/base/kubernetes-manifests.yaml`.

**Nim will add `mk_kustomize_base`**
I am happy to do the work necessary inside `make-release-artifacts.sh` (i.e., implement `mk_kustomize_base`) sometime after we've merged this PR.